### PR TITLE
Travis CI: macOS does not include Python anymore

### PR DESCRIPTION
**Pull request by imatlopez**
##### Checklist

- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

##### Description of change

macOS stopped bundling python, this attempts to bring back travis tests on macOS

### CODE REVIEW

**Comment by cclauss** [comment]
Thanks for doing this but please use __pyenv__ install Python on Travis CI.  It is _way_ faster and supports all Python versions including Python 3.8 which homebrew is still struggling to support.  Try __pyenv install --list__ to see all the versions available.

**Comment by cclauss** [comment]
So Travis is dropping __matrix:__ for __jobs:__ so let's do this around line 4...
```
addons:
  homebrew:
    update: true
    packages:
    - npm
    - pyenv
jobs:
```
~matrix:~

This will ensure that all macOS jobs have acccess to the latest __npm__ and __pyenv__.

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
 cache: pip
-matrix:
+addons:
+  homebrew:
+    update: true
+    packages:
+    - npm
+    - openssl
+    - pyenv
+    - pyenv-virtualenv
```
Comment (cclauss) [inline .travis.yml]: Travis is already creating you a virtualenv...  Layering multiple venvs is not advised.

Comment (imatlopez) [reply .travis.yml]: it should if you have the language set to `python` but that's not available in `osx` no?

Comment (cclauss) [reply .travis.yml]: You are in your own container.  You do not need a venv.

Comment (cclauss) [reply .travis.yml]: Also pyenv will do the openssl thing for you.

Comment (imatlopez) [reply .travis.yml]: I wanted to speed up the openssl, didn't help I realize now :)

**Comment by imatlopez** [comment]
@cclauss got it working, cleaning up. For `pyenv` to work the `PATH` needs to be changed and for python modules to work they need the `python -m` prefix

**Comment by imatlopez** [comment]
@cclauss ready for review :)

**Comment by rvagg** [comment]
I've force pushed out the HEAD commit on master, c392a175f0, so it's in the list of commits here too to be reviewed, edited (needs `test:` as the prefix) and have metadata attached.

**Comment by imatlopez** [comment]
> I've force pushed out the HEAD commit on master, [c392a17](https://github.com/nodejs/node-gyp/commit/c392a175f0a06796861d28eb8b36a9ddd0d41d7d), so it's in the list of commits here too to be reviewed, edited (needs `test:` as the prefix) and have metadata attached.

reworded commit names, is this what you meant?

**Comment by imatlopez** [comment]
Is it me or the python3.5 test is flaky?

**Comment by rvagg** [comment]
> reworded commit names, is this what you meant?

No, sorry, it's not your commit, we'll deal with it when merged but if you rebase onto `master` then you'll be without it. I think it comes from work @cclauss was doing for #1980 but got pushed to master prematurely?

**Comment by imatlopez** [comment]
Do you know why the python 3.5 test is failing? Can it be removed?

**Comment by rvagg** [comment]
nope, will wait for @cclauss' input on that

**Comment by rvagg** [comment]
@cclauss got any suggestions on the 3.5 failure here?

**Comment by cclauss** [comment]
On the 3.5 tests, those seem to be _diff_ tests and my sense is that all the same data is there but not necessarily in the right order.  Perhaps those tests need to look at smaller chunks or sort things before looking at them.

I will study that output in the next few hours...

**Comment by imatlopez** [comment]
What is that flag for? Is it related to using `python -m pytest` vs `pytest`? I know one difference is that the first one sets the current working directory as the `PYTHONPATH` while the second one does not.

**Comment by joaocgreis** [comment]
The 3.5 failure is not related to this PR, so I think this should land when ready.

I saw the 3.5 failure when working on https://github.com/nodejs/node-gyp/pull/1961, but it went away on a second run. That points to something like a difference in Travis runners or a broken cache.

@targos (since I saw you enabled Travis here, but other org owners can also do this) can you clear the Travis cache? Thanks!

**Comment by targos** [comment]
@joaocgreis sure, what should I remove (I don't know anything about Travis caches):

image

**Comment by joaocgreis** [comment]
@targos please delete all repository caches (top right corner). Thanks!

I believe nodejs/node uses the cache to actually make the build possible because of timing issues, but here it should be ok to just delete everything.

**Comment by targos** [comment]
@joaocgreis it's done

**Comment by richardlau** [comment]
I restarted https://travis-ci.com/nodejs/node-gyp/jobs/261487549 after the cache was cleared but it still failed.

**Comment by imatlopez** [comment]
@rvagg anything pending for merging? I saw #1982 and it looks promising but I don't understand the distinction between travis and gh actions.

**Comment by rvagg** [comment]
d6a7e0e1fb9ba4 - @cclauss' commit
345c70e56d2b - the rest, I had a bit of trouble coming up with a summary but I think I got the gist if ot in there

**Comment by rvagg** [comment]
still dealing with linux 3.5 failures https://travis-ci.com/nodejs/node-gyp/builds/141142728, fwiw

**Comment by cclauss** [comment]
See #1995 for more on Python 3.5 failures...

**Pull request closed without merging**